src/main/java/net/dries007/tfc/objects/blocks/BlockPowderKeg.java

```diff
@@ -0,0 +1,460 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.objects.blocks;
+
+import java.util.Random;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.SoundType;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.properties.PropertyBool;
+import net.minecraft.block.state.BlockFaceShape;
+import net.minecraft.block.state.BlockStateContainer;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.projectile.EntityArrow;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.*;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.Explosion;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import net.dries007.tfc.api.capability.size.IItemSize;
+import net.dries007.tfc.api.capability.size.Size;
+import net.dries007.tfc.api.capability.size.Weight;
+import net.dries007.tfc.client.TFCGuiHandler;
+import net.dries007.tfc.objects.te.TEPowderKeg;
+import net.dries007.tfc.util.Helpers;
+
+/**
+ * Powderkeg is an inventory that preserves the contents when sealed
+ * It can be picked up and keeps it's inventory
+ * Sealed state is stored in a block state property, and cached in the TE (for gui purposes)
+ */
+@ParametersAreNonnullByDefault
+public class BlockPowderKeg extends Block implements IItemSize
+{
+    public static final PropertyBool SEALED = PropertyBool.create("sealed");
+    public static final PropertyBool LIT = PropertyBool.create("lit");
```

Comment:
```suggestion
```

Less confusion when it comes to static imports on properties (as they're functionally the same property, so this just simplifies life later).